src/ComposerRepository.php

```diff
@@ -53,6 +64,12 @@ protected function fetchFile($filename, $cacheKey = null, $sha256 = null, $store
             return [];
         }
 
-        return parent::fetchFile($filename, $cacheKey, $sha256, $storeLastModifiedTime);
+        $data = parent::fetchFile($filename, $cacheKey, $sha256, $storeLastModifiedTime);
+
+        if (0 === strpos($filename, 'http://packagist.org/p/symfony/')) {
```

Comment:
Not doing this as this future is unlikely to happen: the jsons are already secured by a sha256 so that enabling https would just make things slower for no benefit on the security side. Running on "http" is a design decision for the packagist infrastructure.

Reply:
it is already possible to force composer to always use HTTPS for packagist (by overriding the packagist.org registration without allowing protocol downgrade), and some people are doing it when they have issues due to a misbehaving HTTP proxy.

Reply:
good to know, updated